### PR TITLE
metadata

### PR DESCRIPTION
**Pull request by LiangJYu**
This PR adds the ability to convert most of the configuration object to a dictionary and write that out to `stdout` (eventually file).

To-do:
~~Write to file~~
~~Support for JSON output~~
~~More documentation~~
~~Add missing fields?~~

### CODE REVIEW

File: src/compass/geo_cslc.py [hunk 1]
```diff
@@ -69,7 +69,7 @@ def run(cfg):
                 f'{output_path}/geo_{burst.burst_id}.tiff',
                 geo_grid.width, geo_grid.length,
                 rdr_burst_raster.num_bands, gdal.GDT_CFloat32,
-                'GTiff')
+                'COG')
```
Comment (vbrancat) [inline src/compass/geo_cslc.py]: Have we included COG as part of `isce3.io.Raster`? I think that if not implemented, specifying a `COG` format might cause Segmentation Fault. Let's still use ENVI and create the COG with `gdal_translate` until a fix get merged in ISCE3

File: src/compass/geo_cslc.py [hunk 2]
```diff
@@ -81,8 +81,8 @@ def run(cfg):
                                       image_grid_doppler, ellipsoid, threshold,
                                       iters,
                                       blocksize, dem_margin, flatten,
-                                      azimuth_carrier=az_carrier_poly2d,
-                                      subraster_bounds=burst_bounds)
```
Comment (vbrancat) [inline src/compass/geo_cslc.py]: Why are we removing the `subraster_bounds` here?

**Review by vbrancat — COMMENTED** [review]
It is a good WIP PR for the metadata and I think we can reuse some of the functionalities for the range/Doppler workflow. We need to dump all this info to file and select a format for it. Also, we would like to write more metadata than the info contained in the runconfig. I am thinking about some burst metadata and of the polygon enclosing the valid metadata. Also, is JSON our final choice for metadata?

To-do:
Write to file
Support for JSON output
More documentation
Add missing fields?

File: src/compass/geo_cslc.py (hunk 1 above)
Comment (LiangJYu) [reply src/compass/geo_cslc.py]: We'll need to use `gdal_translate`. I unsuccessfully attempted a C++ implementation of COG + `isce3.io.Raster`. The new `io.Raster` options for COG did not crash but were unable to write the tiled rasters needed for COG.

File: src/compass/geo_cslc.py (hunk 2 above)
Comment (LiangJYu) [reply src/compass/geo_cslc.py]: I forgot to undo this comment. I was using a different branch and temporarily commented this out in testing.

Comment (LiangJYu) [reply src/compass/geo_cslc.py]: I'll be submitting a PR for GSLC subraster later today so this hopefully this won't be a problem in the near future.

**Review by vbrancat — APPROVED** [review]
Thanks, LGTM

**Pull request merged**